client/src/components/Main.jsx

```diff
@@ -0,0 +1,23 @@
+import React, { Component } from 'react';
+import { Switch, Route } from 'react-router-dom';
+import Entry from './Entry.jsx';
+import Home from './Home.jsx';
+
+class Main extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <main>
+        <Switch>
+          <Route exact path='/' component={Home} />
+          <Route path='/:id' component={Entry} />
+        </Switch>
+      </main>
+    );
+  }
+}
+
+export default Main;
```

Comment:
I hadn't seen react-router-dom before. This is really interesting and a nice way to handle it.

All of the code looks good.